### ISSUE TITLE
--watch exits after first run when using custom schema loader

### ISSUE THREAD

**Issue by jahredhope** [dotansimha/graphql-code-generator#1278]
**Describe the bug**
I'm having a problem getting the `--watch` feature to work and would appreciate some help.

This may be a duplicate of https://github.com/dotansimha/graphql-code-generator/issues/1095
However that issue appears to be solved using a recommended behaviour that doesn't appear to solve my problem.

When running `gql-gen --config codegen.yml --watch` the process ends after the first successful run.
The process logs out: *Watching for changes...* but still exits.
Output:
<img width="295" alt="process exists after run" src="https://user-images.githubusercontent.com/13903378/52325451-2abc5980-2a39-11e9-8e34-c0cac84049ed.png">

I've attempted adding `watch: true` to the schema as-well as removing the documents param as described in the linked issue.

**Removing the below listed custom schema loader fixes the issue.** If I compile my schema modules manually and make that the output of the `src/typeDefs.ts` then remove the custom loader the watch works as expected.
Perhaps I've somehow written my loader wrong?

The loader in question:
```ts
// scripts/load-array-types.ts
export default (schemaString: string) => {
  const typeDefs = require(path.join(process.cwd(), schemaString)).default;
  return makeExecutableSchema({ typeDefs });
};

```

**To Reproduce**
1. Clone the linked proof of concept: https://github.com/jahredhope/apollo-example-dogapi
2. Run `yarn` then `yarn start:generate-types`

**Expected behavior**
I'd expect the process to stay open until killed (e.g. Ctrl + C).

**Schema/Documents**
```yaml
overwrite: true
schema:
  "src/typeDefs.ts":
    loader: "../../../scripts/load-array-types.ts"
generates:
  src/generated-schema-types.ts:
    plugins:
      - "typescript-common"
      - "typescript-server"
      - "typescript-resolvers"
require:
  - ts-node/register
```
The above is a simplified version of the codegen file. The full version can be found here:
https://github.com/jahredhope/apollo-example-dogapi/blob/4f9a48c4a0eb69463fd137173e78cbafc7404cc3/codegen.yml

**Environment:**
 - OS: Mac 
 - Codegen: v0.16.1
 - Node: v10.9.0

**Comment by dotansimha** [comment]
@jahredhope Thank you for reporting it. It's definitely a bug, I think it's because we are not tracking any files in case of custom schema loader. 
I'll try to take a look :)

**Comment by kamilkisiela** [comment]
@jahredhope @dotansimha 

```yaml
# glob
watch: other/files/to/watch/*.ts

# array of files
watch:
 - file-a.ts
 - file-b.ts

# array of globs
watch: 
 - a/*.ts
 - b/*.graphql
```

Is just enough? #1377

**Comment by dotansimha** [comment]
Fixed in 0.18.0 🎉
